src/audio/buffer.c

```diff
@@ -106,16 +108,30 @@ int buffer_set_size(struct comp_buffer *buffer, uint32_t size)
 /* free component in the pipeline */
 void buffer_free(struct comp_buffer *buffer)
 {
+	struct list_item *item;
+	struct list_item *tmp;
+	struct buffer_callback *cb;
+
 	trace_buffer("buffer_free()");
 
+	list_for_item_safe(item, tmp, &buffer->callback_list) {
```

Comment:
We're not removing anything from the list at the moment, but we cannot guarantee that we won't do that in the future.
I've changed that, but we need to be careful if we ever try to remove items from the list, as we might in case of freeing the buffer.